### PR TITLE
Fix memory bloat when displaying plot logs in GUI by sending only new changes

### PR DESCRIPTION
**Pull request by aqk**

### CODE REVIEW

**Comment by aqk** [comment]
Wait to merge until @seeden has integrated the UI side of the change.

File: chia/daemon/server.py [hunk 1]
```diff
@@ -345,38 +346,32 @@ async def _state_changed(self, service: str, state: str):
...
-
-        words = ["Renamed final file"]
-        file_path = config["out_file"]
-        fp = open(file_path, "r")
         while True:
-            new = await loop.run_in_executor(io_pool_exc, fp.readline)
+            new_data = await loop.run_in_executor(io_pool_exc, fp.readline)
 
             if config["state"] is not PlotState.RUNNING:
                 return
```
Comment (wjblanke) [inline chia/daemon/server.py:357]: please add fp.close() here

File: chia/daemon/server.py [hunk 2]
```diff
@@ -345,38 +346,32 @@ async def _state_changed(self, service: str, state: str):
...
+                config["log_new"] = new_data
+                self.state_changed(service_plotter, "log_changed")
 
-            if new:
-                for word in words:
-                    if word in new:
+            if new_data:
+                for word in final_words:
+                    if word in new_data:
                         return
```
Comment (wjblanke) [inline chia/daemon/server.py:367]: please add fp.close() here

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,15 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
 for setuptools_scm/PEP 440 reasons.
 
+
+## UNRELEASED
+
+### Changed
+
+### Fixed
+
+- Fixed excess memory use when displaying plot logs in GUI
```
Comment (arvidn) [inline CHANGELOG.md:16]: I was under the impression that the GUI would request the logs regardless of whether it was showing them or not

**Comment by arvidn** [comment]
@aqk I'm trying to understand how "new_log" is determined. I would expect you to send a cookie back to the GUI of how far it has received (say, a byte-offset into the log file) and the GUI then sending back that offset to ask for all new log entries since then.

However, I don't see any state passed to the GUI to know what's considered "new" in the next call. I also don't see any state in the RPC to remember what was sent last time.

I take it you're assuming a single client requesting the log, is that right?

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**